store/src/java/com/zimbra/cs/mailbox/MailboxCacheManager.java

```diff
@@ -0,0 +1,41 @@
+package com.zimbra.cs.mailbox;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+
+public interface MailboxCacheManager {
+
+    Collection<Integer> getMailboxIds();
+
+    Set<String> getAccountIds();
+
+    Set<Map.Entry<String, Integer>> getMailboxIdsByAccountId();
+
+    int getMailboxCount();
+
+    Integer getMailboxKey(String accountId);
+
+    void removeMailboxId(String accountId);
+
+    Object retrieveFromCache(int mailboxId, boolean trackGC, MailboxManager mailboxManager) throws MailServiceException;
+
+    void cacheMailbox(int mailboxId, MailboxMaintenance maint);
+
+    Object getMailbox(int mailboxId);
```

Comment:
Why is the return type `Object` instead of `Mailbox`?

Reply:
Changed to `Mailbox`